Derivs/src/derivs.cxx

```diff
@@ -87,11 +101,50 @@ calc_derivs(const GF3D5<T> &gf, const vec<GF3D5<T>, dim> &dgf,
         });
     break;
 
+  case 8:
+    grid.loop_int_device<CI, CJ, CK, vsize>(
+        grid.nghostzones,
+        [=] CCTK_DEVICE(const PointDesc &p) CCTK_ATTRIBUTE_ALWAYS_INLINE {
+          const vbool mask = mask_for_loop_tail<vbool>(p.i, p.imax);
+          // Take ghost points into account
+          const vbool mask1 = mask_for_loop_tail<vbool>(p.i, p.imax + 8 / 2);
+          const GF3D5index index(layout, p.I);
+          const auto val = gf0(mask, p.I);
+          const auto dval = calc_deriv<8>(gf0, mask1, p.I, dx);
+          gf.store(mask, index, val);
+          dgf.store(mask, index, dval);
+        });
+    break;
+
   default:
     CCTK_VERROR("Unsupported derivative order %d", deriv_order);
   }
 }
 
+template <int CI, int CJ, int CK, typename T>
+CCTK_ATTRIBUTE_NOINLINE void
+calc_derivs(const vec<GF3D5<T>, dim> &gf,
+            const vec<vec<GF3D5<T>, dim>, dim> &dgf, const GF3D5layout layout,
+            const GridDescBaseDevice &grid, const vec<GF3D2<const T>, dim> &gf0,
+            const vect<T, dim> dx, const int deriv_order) {
+  for (int a = 0; a < dim; ++a)
+    calc_derivs<CI, CJ, CK>(gf(a), dgf(a), layout, grid, gf0(a), dx,
+                            deriv_order);
+}
```

Comment:
these are new as well. Derivs used to only take plain `GF3D5` as inputs.

Reply:
this was also there before. What I did is make it more complete and consistent